q2_demux/_summarize/assets/src/plot.js

```diff
@@ -0,0 +1,229 @@
+    .height(props.height - props.margin.bottom - props.margin.top)
+    .domain([0, 100]);
+
+  const brush = d3.brush();
+  let idleTimeout;
+  const idleDelay = 350;
+
+  const idled = () => {
+    idleTimeout = null;
+  };
+
+  const zoom = () => {
+    const t = svg.transition().duration(750);
+    svg.select('.axis--x').transition(t).call(xAxis);
+    svg.select('.axis--y').transition(t).call(yAxis);
+    svg.selectAll('.boxplot').transition(t)
+      .call(chart.domain(y.domain())
+                 .width((x.range()[1] - x.range()[0]) / (x.domain()[1] - x.domain()[0]) / 2))
+      .attr('transform', d => `translate(${x(d[0])}, ${props.margin.top})`);
+  };
+
+  const brushEnded = () => {
+    const s = d3.event.selection;
+    if (!s) {
+      if (!idleTimeout) {
+        idleTimeout = setTimeout(idled, idleDelay);
+        return idleTimeout;
+      }
+      x.domain(x0);
+      y.domain(y0);
+    } else {
+      x.domain([s[0][0], s[1][0]].map(x.invert, x));
+      y.domain([s[1][1], s[0][1]].map(y.invert, y));
+      svg.select('.brush').call(brush.move, null);
+    }
+    return zoom();
+  };
+
+  brush.on('end', brushEnded);
+
+  svg.append('g')
+      .attr('class', 'brush')
+      .attr('z-index', -1)
+      .call(brush);
+
+  svg.selectAll('.box')
+    .attr('font', '10px sans-serif')
+    .data(data)
+  .enter()
+    .append('g')
+    .attr('class', 'boxplot')
+    .attr('transform', d => `translate(${x(d[0])}, ${props.margin.top})`)
+    .call(chart.width((x.range()[1] - x.range()[0]) / (x.domain()[1] - x.domain()[0]) / 2));
+
+  svg.append('rect')
+      .attr('width', props.width + props.margin.left + props.margin.right)
+      .attr('height', props.margin.bottom * 4)
+      .attr('y', props.height - props.margin.bottom)
+      .attr('fill', 'white');
+
+  svg.append('rect')
+      .attr('width', props.margin.left * 4)
+      .attr('x', -props.margin.left * 3)
+      .attr('height', props.height + 200)
+      .attr('fill', 'white');
+
+  svg.append('rect')
+    .attr('width', props.margin.right + 1000)
+    .attr('height', props.height + 200)
+    .attr('x', props.width + 10)
+    .attr('fill', 'white');
+
+  svg.append('g')
+      .attr('class', 'axis axis--x')
+      .attr('transform', `translate(0, ${(props.height - props.margin.bottom)})`)
+      .call(xAxis);
+
+  svg.append('g')
+      .attr('class', 'axis axis--y')
+      .attr('transform', `translate(${props.margin.left},0)`)
+      .call(yAxis);
+
+  svg.selectAll('.domain')
+      .style('display', 'none');
+
+  svg.append('text')
+      .attr('transform', 'rotate(-90)')
+      .attr('x', 0 - (props.height / 2))
+      .attr('dy', '1em')
+      .attr('font-size', '10px')
+      .style('text-anchor', 'middle')
+      .text('Quality Score');
+
+  svg.append('text')
+      .attr('x', props.width / 2)
+      .attr('y', props.height)
+      .attr('dy', '1em')
+      .attr('font-size', '10px')
+      .style('text-anchor', 'middle')
+      .text('Sequence Base');
```

Comment:
I have some magic white rectangles boxing in the plot, otherwise on zooming it shows beyond the axis, so this will have to be added after those blocks in order to be visible, and those blocks will have to be after the boxes in order to block them, since `svg` is entirely order based making this stuck as-is.